README.md

```diff
@@ -0,0 +1,93 @@
+
+The BookShop project is a web-based REST application developed using Java Spring. It provides a platform for both shoppers and managers to interact with books, shopping carts, orders, and inventory. The project will be developed incrementally, with features broken down into manageable parts.
+
+## 2. Entities <a name="entities"></a>
+
+The following domain models (entities) are central to this application:
+
+- **User**: Represents registered users with authentication details and personal information.
+- **Role**: Defines user roles such as admin, manager or regular user.
+- **Book**: Represents books available in the store.
+- **Category**: Classifies books into different categories.
+- **ShoppingCart**: Represents a user's shopping cart.
+- **CartItem**: Represents an item in a user's shopping cart.
+- **Order**: Represents an order placed by a user.
+- **OrderItem**: Represents an item in a user's order.
+
+## 3. Features <a name="features"></a>
+### Shoppers Can:
+
+- **Join and Sign In**: Users can register and sign in to browse and purchase books.
+- **Browse Books**: Users can view all available books, explore individual book details.
+- **Bookshelf Sections**: Users can explore bookshelf sections and view all books within a specific section.
+- **Use the Basket**: Users can add books to their shopping basket, view its contents, and remove books from it.
+- **Purchase Books**: Users can buy all the books in their basket and view their order history.
+
+### Managers Can:
+
+- **Manage Books**: Managers can add new books to the store, update book details, and remove books from the inventory.
+- **Organize Sections**: Managers can create, modify, or delete bookshelf sections to organize the inventory.
+- **Manage Receipts**: Managers can change the status of order receipts, such as marking them as “Shipped” or “Delivered”.
+
+### Admins Can:
+- **Manage Users**: Admins can create users with specific role
+
+## 4. Getting Started <a name="getting-started"></a>
+
+To get started with my Online Book Store project, follow these steps:
+
+1. **Clone the Repository**: Clone this GitHub repository to your local machine.
+
+2. **Install Dependencies**: Ensure you have Java Spring and necessary dependencies installed.
+
+3. **Database Configuration**: Configure the database connection settings in the application properties.
+
+4. **Run the Application**: Start the Spring application and access it through your web browser.
+
+## 5. Docker <a name="docker"></a>
+
+Project also offers a Docker containerized version of this application. To run it using Docker, follow these additional steps:
+
+1. **Docker Setup**: Install Docker on your machine if not already installed.
+
+2. **Add necessary dependency**: Add this lines at pom.xml in <dependencies> section:
+
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-docker-compose</artifactId>
```

Comment:
this dependency only needed for local development when runnin program via main method. It is not required for using docker commands